esadapter/src/main/java/com/ncc/neon/adapters/es/ElasticsearchResultsConverter.java

```diff
@@ -94,7 +100,26 @@ public static TabularQueryResult convertResults(Query query, SearchResponse resp
         return results;
     }
 
-    private static List<Map<String, Object>> extractHitsFromResults(SearchResponse response) {
+    public static List<Map<String, Object>> getScrolledResults(Scroll scroll, SearchResponse response, RestHighLevelClient client) throws IOException {
+        String scrollId = response.getScrollId();
+        List<Map<String, Object>> allHits = new ArrayList<>();
+        allHits.addAll(ElasticsearchResultsConverter.extractHitsFromResults(response));
+
+        while (response.getHits().getHits() != null && response.getHits().getHits().length > 0) {
```

Comment:
@jacklechnercaci Beside what you already said about only using the search scroll API if the limit is greater than 10K, we should also only return a number of results equal to or less than the limit.  For example, if my limit is 15K, and I have 20K records in my datastore, currently this method returns all 20K records, and not just the first 15K.

Reply:
- Updated the PR so that if the limit is over 10k, it does a scrolled request.
- Now returns the number of records equal to the limit.